### PR TITLE
Ie11 transitioning node

### PR DESCRIPTION
**Pull request by nathanbowser**
Recreating this so I can comment on some things...

### CODE REVIEW

File: lib/enter.js [hunk 1]
```diff
@@ -16,7 +16,7 @@ module.exports = function (tree) {
     if (transitions) {
       enter.classed('transitioning-node incoming-node', true)
       tree._forceRedraw() // Force a redraw so we see the updates transitioning
-      enter.classed('transitioning-node incoming-node', false)
+      enter.classed('transitioning-node', false)
     }
```
Comment (nathanbowser) [inline lib/enter.js]: Now that you're no longer removing the class `incoming-node`, it's going to stick around forever, which doesn't make sense. 

screen shot 2016-12-07 at 4 07 21 pm

File: tree.scss [hunk 2]
```diff
@@ -66,31 +66,6 @@ $timing-fn: ease-out;
...
-    /*
-     * In IE11 (oldest IE version supported by the tree), there's a bug
-     * and incoming nodes don't transition properly if the opacity isn't 1.
-     * So this disables the fade-in effect in IE11, so the translate transformation
-     * looks good.
-     */
-    .tree {
-      &.tree-overflow {
-        overflow-y: scroll;
-      }
```
Comment (nathanbowser) [inline tree.scss]: Why is this removed? overflow-y on the tree is for forcing IE11 to have scrollbars if this `tree-overflow` is set. That should be unrelated to this issue.

File: lib/enter.js (hunk 1 above)
Comment (derekstory) [inline lib/enter.js]: Actually, now that I'm looking at this again after a break from it. That entire class isn't even necessary. We can just apply a default transition to the transitioning node. It doesn't need an opacity set because the default is 1. Updated.

File: tree.scss (hunk 2 above)
Comment (derekstory) [inline tree.scss]: It shouldn't have been. Added back.

File: tree.scss [hunk 3]
```diff
@@ -179,8 +162,17 @@ $timing-fn: ease-out;
...
       ul {
         li.node {
+          @include transition-property(transform, opacity, height);
           &.transitioning-node {
+            // Initially set the opacity to 0 for incoming nodes so they can transition to 1. 
+            // Set to 0 for outgoing nodes
             opacity: 0;
+            // Don't apply transition on opacity here. Otherwise, IE11 has a bug that will break the entire transition
+            // Instead, apply the transition to the incoming/outgoing classes. 
+            @include transition-property(height, transform);
```
Comment (derekstory) [inline tree.scss]: @nathanbowser This line is pretty much the only actual thing that changed from how it originally worked. And that is only for a split second since its applied and instantly removed.

File: tree.scss [hunk 4]
```diff
@@ -333,6 +324,7 @@ $timing-fn: ease-out;
       li.node:not(.transient):not(.selected):not(.transitioning-node) {
         .node-contents, .label-mask:after {
           opacity: .6;
+          @include transition(none);
```
Comment (nathanbowser) [inline tree.scss]: Is this still needed?

File: tree.scss (hunk 3 above)
Comment (derekstory) [inline tree.scss]: No, because outgoing-node needs that `opacity: 0`.

Comment (derekstory) [inline tree.scss]: And the answer to how it fixes it is...it shouldn't. But its IE11 so who knows..

Comment (nathanbowser) [inline tree.scss]: Oh, right! How about this:

```
          &.transitioning-node {
            // Initially set the opacity to 0 for incoming nodes so they can transition to 1. 
            // Set to 0 for outgoing nodes
            opacity: 0;
            
            &:not(.outgoing-node) {
              // Don't apply transition on opacity here. Otherwise, IE11 has a bug that will break the entire transition
              // Instead, apply the transition to the incoming/outgoing classes. 
              @include transition-property(height, transform); 
            }
          }
```

File: tree.scss (hunk 4 above)
Comment (derekstory) [inline tree.scss]: Yeah, added a comment in. That opacity causes the same weird transition bug in ie11. No idea why.

File: tree.scss (hunk 3 above)
Comment (derekstory) [inline tree.scss]: I like it. Updated.

**Comment by nathanbowser** [comment]
This is at a point, where I don't think I can break it, and the code is cleaner. I'm still just terrified because I don't get it.

One thing I'm not sure if this breaks... When a tree goes to the editable state, should the opacity animate or is it instant?

**Comment by derekstory** [comment]
In SB3 it looks like it is instant so I assume thats how it should be?
